tuxemon/states/pc/__init__.py

```diff
@@ -188,3 +203,158 @@ def initialize_items(self) -> Generator[MenuItem[None], None, None]:
             item = MenuItem(label, None, None, None)
             item.enabled = False
             yield item
+
+
+class MonsterTakeState(PygameMenuState):
+    """Menu for the Monster Take state.
+
+    Shows all tuxemon currently in a storage kennel, and selecting one puts it
+    into your current party."""
+
+    def add_menu_items(
+        self: MonsterTakeState,
+        menu: pygame_menu.Menu,
+        items: Sequence[Tuple[str, MenuGameObj]],
+    ) -> None:
+
+        width, height = prepare.SCREEN_SIZE
+
+        for monster, callback in items:
+            label = T.translate(monster.name).upper()
+            new_button = menu.add.button(label, callback)
+            results = db.lookup(monster.slug, table="monster").dict()
+            new_image = menu.add.image(
+                graphics.transform_resource_filename(
+                    results["sprites"]["menu1"] + ".png"
+                )
+            )
+            new_image.scale(prepare.SCALE, prepare.SCALE)
+
+    def startup(self, box_name: String, **kwargs: Any) -> None:
+        width, height = prepare.SCREEN_SIZE
+
+        theme = get_theme()
+        theme.scrollarea_position = pygame_menu.locals.POSITION_EAST
+        columns = 3
+
+        self.box_name = box_name
+        self.player = local_session.player
+        self.box = self.player.monster_boxes[self.box_name]
+
+        num_mons = len(self.box)
+        # Widgets are like a pygame_menu label, image, etc.
+        num_widgets_per_monster = 2
+        rows = int(num_mons * num_widgets_per_monster / columns) + 1
+        # Make sure rows are divisible by num_widgets
+        while rows % num_widgets_per_monster != 0:
+            rows += 1
+
+        super().startup(
+            height=height, width=width, columns=columns, rows=rows, **kwargs
+        )
+
+        def take_monster(monster: monster) -> None:
```

Comment:
Are you sure this typehint is right?

Reply:
Probably need to do the following:

Add `TYPE_CHECKING` to list of typing imports

Then after the imports:

`if TYPE_CHECKING:
    from wherever import Monster
`